src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -310,6 +310,13 @@ private async ValueTask<EvaluationEvent> EvaluateFeature<TContext>(string featur
                     }
                     else
                     {
+                        if (targetingContext == null)
+                        {
+                            Logger?.LogWarning(useContext ?
+                                $"The {nameof(TargetingContext)} required for variant assignment was not provided." :
+                                $"The {nameof(TargetingContext)} required for variant assignment was not retrieved from the {nameof(ITargetingContextAccessor)}.");
```

Comment:
`The {nameof(TargetingContext)} required for variant assignment was not retrieved from the {nameof(ITargetingContextAccessor)}.`

What happens if the user hasn't even registered an `ITargetingContextAccessor`? Is this the only warning they'll get? If so, that would be confusing.

Reply:
`was not retrieved`

That wording is a bit confusing to me. It seems like it could mean all three of these things:

* We didn't even attempt to get it
* The accessor wasn't available
* We used a provided accessor, but it returned no targeting context.